src/runtime-rs/crates/resource/src/network/network_info/network_info_from_link.rs

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::{convert::TryFrom, net::Ipv4Addr};
+
+use agent::{ARPNeighbor, IPAddress, IPFamily, Interface, Route};
+use anyhow::{Context, Result};
+use async_trait::async_trait;
+use futures::stream::TryStreamExt;
+use netlink_packet_route::{
+    self, neighbour::NeighbourMessage, nlas::neighbour::Nla, route::RouteMessage,
+};
+
+use super::NetworkInfo;
+use crate::network::utils::{address::Address, link};
+
+#[derive(Debug)]
+pub(crate) struct NetworkInfoFromLink {
+    interface: Interface,
+    neighs: Vec<ARPNeighbor>,
+    routes: Vec<Route>,
+}
+
+impl NetworkInfoFromLink {
+    pub async fn new(
+        handle: &rtnetlink::Handle,
+        link: &dyn link::Link,
+        hw_addr: &str,
+    ) -> Result<Self> {
+        let attrs = link.attrs();
+        let name = &attrs.name;
+        let mut addr_msg_list = handle
+            .address()
+            .get()
+            .set_link_index_filter(attrs.index)
+            .execute();
+
+        let mut addresses = Vec::new();
+        while let Some(addr_msg) = addr_msg_list.try_next().await? {
+            if addr_msg.header.family as i32 != libc::AF_INET {
+                warn!(sl!(), "unsupported ipv6 addr. {:?}", addr_msg);
+                continue;
+            }
+            let a = Address::try_from(addr_msg).context("get addr from msg")?;
+            if a.addr.is_loopback() {
+                continue;
+            }
+
+            addresses.push(IPAddress {
+                family: if a.addr.is_ipv4() {
+                    IPFamily::V4
+                } else {
+                    IPFamily::V6
+                },
+                address: a.addr.to_string(),
+                mask: format!("{}", a.perfix_len),
```

Comment:
a.perfix_len.to_string() is better

Reply:
Thanks for your review, has been updated.